drivers/wifi/nrf700x/zephyr/src/zephyr_net_if.c

```diff
@@ -171,6 +171,12 @@ static void ip_maddr_event_handler(struct net_mgmt_event_callback *cb,
 		return;
 	}
 
+	/* TODO: Handle multiple vendor Wi-Fi chipsets */
+	if (!strncmp(dev->name, "wlan", 4)) {
```

Comment:
Instead of relying on device name (which as noted in TODO have its downsides), perhaps it'd make more sense to store the interface pointer/index in `wifi_nrf_if_init_zep()` and just compare here?

Reply:
But the problem here is that to access the driver structures we need to know if `dev->data` is owned by `nrf700x` driver.

Reply:
Ah I was referring to storing iface pointer in a nrf700x data structures, but you are referring to a global which I wanted to avoid, esp. as we will introducing multiple interfaces in the feature.

Reply:
I had renamed wlan0->nrfwifi0 to future safe but that was invasive, so, went with this workaround. There is a pending todo to evaluate better option than using `dev->name` in samples to get iface, so, we can club this to that.

Reply:
Went with global variable based check as we were already doing it elsewhere.